tiledb/sm/array_schema/dimension.cc

```diff
@@ -1588,21 +1588,45 @@ Status Dimension::set_null_tile_extent_to_range() {
     return LOG_STATUS(Status::DimensionError(
         "Cannot set tile extent to domain range; Domain not set"));
 
+  typedef typename std::make_unsigned<T>::type unsigned_t;
```

Comment:
Nit: I'd call this `unsigned_T`, which you'd use if you had two template arguments `T` and `U`.
Nit: I'd move this to the line immediately before use.